pkg/runner/flags.go

```diff
@@ -225,3 +246,52 @@ func validate(cfg *Config) error {
 	}
 	return nil
 }
+
+func addAliases(fs *flag.FlagSet, aliases map[string][]string, descSuffix string) {
+	aliasDescFormat := aliasDescTemplate + descSuffix
+	set := false
+	fs.VisitAll(func(f *flag.Flag) {
+		if flagAliases, ok := aliases[f.Name]; ok {
+			set = false
+			for _, alias := range flagAliases {
+				set = true
+				fs.Var(f.Value, alias, fmt.Sprintf(aliasDescFormat, f.Name))
+			}
+
+			if !set {
+				panic(fmt.Sprintf("trying to set an flag alias for a flag that is missing: %s", f.Name))
+			}
+		}
+	})
+}
+
+func flagAliasesUsed(fs *flag.FlagSet) map[string]string {
+	var (
+		result      = make(map[string]string)
+		newFlagName string
+		count       int
+	)
+
+	fs.Visit(func(f *flag.Flag) {
+		if strings.Contains(f.Usage, deprecatedFlagSuffix) {
+			i, err := fmt.Sscanf(f.Usage, aliasDescTemplate, &newFlagName)
+
+			switch {
+			case f.Name == "migrate":
+				// Migrate flag is a special case since it's not just a rename
+				// of an old flag. We handle it separately when logging
+				// use of deprecated flags.
+				return
+			case err != nil:
+				fallthrough
+			case i != 1:
```

Comment:
shouldn't this be `i < 1` as more than one deprecated flags can be passed?

Reply:
Actually no, an alias can have only one flag that it is aliasing.